datasetinsights/io/downloader/http_downloader.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import os
+import zipfile
+
+from datasetinsights.io.download import (
+    download_file,
+    get_checksum_from_file,
+    validate_checksum,
+)
+from datasetinsights.io.downloader.base import DatasetDownloader
+from datasetinsights.io.exceptions import ChecksumError
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPDownloader(DatasetDownloader, protocol="http://"):
+    """ This class is used to download data from any HTTP or HTTPS public url
+        and perform function such as downloading the dataset and checksum
+        validation.
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def download(self, source_uri, output, checksum_file=None, **kwargs):
+        """ This method is used to download the dataset from HTTP or HTTPS url.
+
+        Args:
+            source_uri (str): This is the downloader-uri that indicates where
+                              the dataset should be downloaded from.
+
+            output (str): This is the path to the directory where the download
+                          will store the dataset.
+
+            checksum_file (str): This is path of the txt file that contains
+                                 checksum of the dataset to be downloaded. It
+                                 can be HTTP or HTTPS url or local path.
+
+        Raises:
+            ChecksumError: This will raise this error if checksum doesn't
+                           matches
+
+        """
+
+        dataset_path = os.path.join(output, "dataset.zip")
+        download_file(source_uri, dataset_path)
+
+        if checksum_file:
+            logger.debug("Reading checksum from checksum file.")
+            checksum = get_checksum_from_file(checksum_file)
+            try:
+                logger.debug("Validating checksum!!")
+                validate_checksum(dataset_path, checksum)
+            except ChecksumError as e:
+                logger.info("Checksum mismatch. Deleting the downloaded file.")
+                os.remove(dataset_path)
+                raise e
+
+        self.unzip_file(dataset_path, output)
```

Comment:
I've merged this branch to my local branch. I figured out that this would unzip it to a subfolder. 
For example, if we set output = "/data/groceries/v3". It would download a "dataset.zip" in the "/data/groceries/v3" and unzip it to "/data/groceries/v3/v3".

Reply:
Yes, that's because groceries dataset zip file contains a directory `v3` which in turn contains the dataset files. I don't think it should be the responsibility of downloader to find the files or get rid of directories, instead all the dataset files should be zipped directly instead of a zipping a directory that contains those files.

Reply:
I will zip the GroceriesReal directly and update on GCS.